centreon/www/widgets/host-monitoring/src/index.php

```diff
@@ -366,16 +424,26 @@
     }
 
     if (isset($preferences['display_last_comment']) && $preferences['display_last_comment']) {
-        $res2 = $dbb->prepare(
-            'SELECT data FROM comments where host_id = :hostId
-            AND service_id = 0 ORDER BY entry_time DESC LIMIT 1'
-        );
-        $res2->bindValue(':hostId', $row['host_id'], \PDO::PARAM_INT);
-        $res2->execute();
-        if ($row2 = $res2->fetch()) {
-            $data[$row['host_id']]['comment'] = substr($row2['data'], 0, $commentLength);
-        } else {
-            $data[$row['host_id']]['comment'] = '-';
+        try {
+            $query = <<<SQL
+                SELECT data FROM comments where host_id = :hostId
+                AND service_id = 0 ORDER BY entry_time DESC LIMIT 1
+            SQL;
+            $res2 = $dbb->prepare($query);
+            $res2->bindValue(':hostId', $row['host_id'], PDO::PARAM_INT);
+            $res2->execute();
+            if ($row2 = $res2->fetch()) {
+                $data[$row['host_id']]['comment'] = substr($row2['data'], 0, $commentLength);
+            } else {
+                $data[$row['host_id']]['comment'] = '-';
+            }
```

Comment:
[Nitpick & subjective sharing]

As much as possible I try to :
- have meaningful naming for vars
- avoid if else blocks & replace by ternary format as I find it easier to read.

```suggestion
        $commentRow = $res2->fetch();
        $data[$row['host_id']]['comment'] = $commentRow
            ? substr($commentsRows['data'], 0, $commentLength)
            : '-';
```

Reply:
Yes, you're right! But I only caught the exception to log it, the code is not my production :) I have two revisions, so I won't be making that change. I'm solving this comment.